IOPool/Streamer/src/StreamerInputFile.cc

```diff
@@ -140,13 +178,14 @@ namespace edm {
   void StreamerInputFile::readStartMessage() {
     using namespace edm::storage;
     IOSize nWant = sizeof(HeaderView);
-    IOSize nGot = readBytes(&headerBuf_[0], nWant);
+    char* headerPtr = &headerBuf_[0];
+    IOSize nGot = readBytes(&headerPtr, nWant, false);
     if (nGot != nWant) {
       throw Exception(errors::FileReadError, "StreamerInputFile::readStartMessage")
           << "Failed reading streamer file, first read in readStartMessage\n";
     }
 
-    HeaderView head(&headerBuf_[0]);
+    HeaderView head(headerPtr);
```

Comment:
Could I suggest to enclose this and the following lines into a code block so that it is clear that `head` can not be used after the possible  `headerBuf_.resize()`? I.e.
```cpp
    {
      HeaderView head(headerPtr);
      uint32 code = head.code();
      if (code != Header::INIT) /** Not an init message should return ******/
      {
        throw Exception(errors::FileReadError, "StreamerInputFile::readStartMessage")
            << "Expecting an init Message at start of file\n";
        return;
      }
    }
```

(same comment for the similar pattern in `readEventMessage()`)